common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Configuration.java

```diff
@@ -24,18 +24,73 @@
 
 import android.net.Uri;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.microsoft.identity.common.exception.ServiceException;
+import com.microsoft.identity.common.internal.logging.Logger;
 import com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryOAuth2Configuration;
+import com.microsoft.identity.common.internal.providers.oauth2.OpenIdProviderConfiguration;
+import com.microsoft.identity.common.internal.providers.oauth2.OpenIdProviderConfigurationClient;
 
 import java.net.URL;
 
 public class MicrosoftStsOAuth2Configuration extends AzureActiveDirectoryOAuth2Configuration {
 
+    private static final String TAG = MicrosoftStsOAuth2Configuration.class.getSimpleName();
+
+    private static final String ENDPOINT_VERSION = "v2.0";
+    private static final String FALLBACK_ENDPOINT_SUFFIX = "/oAuth2/v2.0";
+    private static final String FALLBACK_AUTHORIZE_ENDPOINT_SUFFIX = FALLBACK_ENDPOINT_SUFFIX + "/authorize";
+    private static final String FALLBACK_TOKEN_ENDPOINT_SUFFIX = FALLBACK_ENDPOINT_SUFFIX + "/token";
+
+    /**
+     * Get the authorization endpoint to be used for making a authorization request.
+     * This must NOT be called from the main thread.
+     *
+     * @return URL the authorization endpoint
+     */
     public URL getAuthorizationEndpoint() {
-        return getEndpoint(getAuthorityUrl(), "/oAuth2/v2.0/authorize");
+        final OpenIdProviderConfiguration openIdConfig = getOpenIdWellKnownConfigForAuthority();
+        if (openIdConfig != null) {
+            return getEndpoint(openIdConfig.getAuthorizationEndpoint());
+        }
+        return getEndpoint(getAuthorityUrl(), FALLBACK_AUTHORIZE_ENDPOINT_SUFFIX);
     }
 
+    /**
+     * Get the token endpoint to be used for making a token request.
+     * This must NOT be called from the main thread.
+     *
+     * @return URL the token endpoint
+     */
     public URL getTokenEndpoint() {
-        return getEndpoint(getAuthorityUrl(), "/oAuth2/v2.0/token");
+        final OpenIdProviderConfiguration openIdConfig = getOpenIdWellKnownConfigForAuthority();
+        if (openIdConfig != null && openIdConfig.getTokenEndpoint() != null) {
+            return getEndpoint(openIdConfig.getTokenEndpoint());
+        }
+        return getEndpoint(getAuthorityUrl(), FALLBACK_TOKEN_ENDPOINT_SUFFIX);
+    }
+
+    @Nullable
+    private URL getEndpoint(@NonNull final String endpoint) {
```

Comment:
nit : rename method for clarity, also there is a method with a same name (from before) in this class , which lacks clarity too, these can get confusing

Reply:
Addressed